docs/workers.md

```diff
@@ -242,40 +204,142 @@ endpoints matching the following regular expressions:
+ensure that the purge history admin API is not used while pagination requests
+for the room are in flight:
+
+    ^/_matrix/client/(api/v1|r0|unstable)/rooms/.*/messages$
+
+Note that the `client` and `federation` must be added to the listener resources
+in the worker config.
+
+
+#### Load balancing
+
+Multiple instances of this app can be run and requests load balanced between
+them. However, different endpoints have different characteristics and so admins
+may wish to run multiple groups of workers handling different endpoints so that
+load balancing can be done in different ways.
 
-The `^/_matrix/federation/v1/send/` endpoint must only be handled by a single
-instance.
+For `/sync` and `/initialSync` requests it will be more efficient if all
+requests from a particular user are routed to a single instance. Extracting a
+user ID from the access token or `Authorization` header is currently left as an
+exercise for the reader. Admins may additionally wish to separate out `/sync`
+requests that have a `since` query parameter from those that don't (and
+`/initialSync`), as requests that don't are known as "initial sync" that happens
+when a user logs in on a new device and can be *very* resource intensive, so
+isolating these requests will stop them from interfering with other users ongoing
+syncs.
 
-Note that `federation` must be added to the listener resources in the worker config:
+Federation and client requests can be balanced via simple round robin.
+
+The inbound federation transaction request `^/_matrix/federation/v1/send/`
+should be balanced by source IP so that transactions from the same remote server
+go to the same process.
+
+Registration/login requests can be handled separately purely to help ensure that
+unexpected load doesn't effect new logins and sign ups.
+
+Finally, event sending requests can be balanced by the embedded room ID (or
+URI, or even just round robin). If there is a large bridge connected that is
+sending or may send lots of events, then a dedicated set of workers can be
+provisioned to ensure that bursts or increases of event sending is isolated from
+effecting events sent by real users.
+
+#### Stream writers
+
+Additionally, there is *experimental* support for moving writing of specific
+streams (such as events and typing) off of the main process to a particular
+worker. This requires use of Redis.
+
+To enable this then the worker must have a HTTP replication listener configured,
+have a `worker_name` and be listed in the `instance_map` config. For example to
+move event persistence off to a dedicated worker, the shared configuration would
+include:
 
 ```yaml
-worker_app: synapse.app.federation_reader
-...
-worker_listeners:
- - type: http
-   port: <port>
-   resources:
-     - names:
-       - federation
+instance_map:
+    event_persister1:
+        host: localhost
+        port: 8034
+
+streams_writers:
+    events: event_persister1
```

Comment:
which streams can be moved in this way? it says "such as events and typing" above but I'm not sure if that is complete?